applications/OptimizationApplication/python_scripts/model_part_controllers/connectivity_preserving_model_part_duplication_controller.py

```diff
@@ -0,0 +1,39 @@
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.OptimizationApplication.model_part_controllers.model_part_controller import ModelPartController
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, _) -> ModelPartController:
+    if not parameters.Has("settings"):
```

Comment:
isn't it supposed to be `Settings` (note the capitalization)? I'm not fond of pointless capitalization either, but we should be consistent with core.